src/client/Client.js

```diff
@@ -780,4 +796,48 @@ export default class Client {
     get isClientShutDown() {
         return this._isShutdown;
     }
+
+    /**
+     * Validates that all nodes in a network are in the same shard and realm.
+     *
+     * @param {{[key: string]: (string | AccountId)}} network
+     */
+    static _validateNetworkConsistency(network) {
+        let shard = undefined;
+        let realm = undefined;
+
+        for (const accountId of Object.values(network)) {
+            const accountIdStr = accountId.toString();
+            const [nodeShard, nodeRealm] = accountIdStr.split(".").map(Number);
```

Comment:
Rename the variables to `currentShard, currentRealm` and rename `shard` and `realm` to `lastShard` and `lastRealm`??

Reply:
also [array.every is a perfect fit here ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)